airbyte-integrations/connectors/destination-firebolt/destination_firebolt/destination.py

```diff
@@ -0,0 +1,145 @@
+                    config["loading_method"]["aws_key_id"],
+                    config["loading_method"]["aws_key_secret"],
+                    config["loading_method"]["s3_region"],
+                )
+            else:
+                logger.info("Using the SQL writing strategy")
+                writer = FireboltSQLWriter(connection)
+
+            for configured_stream in configured_catalog.streams:
+                if configured_stream.destination_sync_mode == DestinationSyncMode.overwrite:
+                    writer.delete_table(configured_stream.stream.name)
+                    logger.info(f"Stream {configured_stream.stream.name} is wiped.")
+                writer.create_raw_table(configured_stream.stream.name)
+
+            for message in input_messages:
+                if message.type == Type.STATE:
+                    yield message
+                elif message.type == Type.RECORD:
+                    data = message.record.data
+                    stream = message.record.stream
+                    # Skip unselected streams
+                    if stream not in streams:
+                        logger.debug(f"Stream {stream} was not present in configured streams, skipping")
+                        continue
+                    writer.queue_write_data(stream, str(uuid4()), datetime.now(), json.dumps(data))
+
+            # Flush any leftover messages
+            writer.flush()
+
+    def check(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> AirbyteConnectionStatus:
+        """
+        Tests if the input configuration can be used to successfully connect to the destination with the needed permissions
+            e.g: if a provided API token or password can be used to connect and write to the destination.
+
+        :param logger: Logging object to display debug/info/error to the logs
+            (logs will not be accessible via airbyte UI if they are not passed to this logger)
+        :param config: Json object containing the configuration of this destination, content of this json is as specified in
+        the properties of the spec.json file
+
+        :return: AirbyteConnectionStatus indicating a Success or Failure
+        """
+        try:
+            with establish_connection(config, logger) as connection:
+                # We can only verify correctness of connection parameters on execution
+                with connection.cursor() as cursor:
+                    cursor.execute("SELECT 1")
+                # Test access to the bucket
+                if config["loading_method"]["method"] == "S3":
+                    FireboltS3Writer(
+                        connection,
+                        config["loading_method"]["s3_bucket"],
+                        config["loading_method"]["aws_key_id"],
+                        config["loading_method"]["aws_key_secret"],
+                        config["loading_method"]["s3_region"],
+                    )
```

Comment:
@ptiurin could you move this into firebolt.py/database.py and get the writer from there. This should avoid creating lengthy initiations every place and also streamline the database connection part. 

may be something like Firebolt(**config) and you can create a connection and write inside that class

Reply:
I've refactored writer instantiation out of destination.py